telemetry/src/lib.rs

```diff
@@ -0,0 +1,84 @@
+use std::io::IsTerminal as _;
+
+use tracing::level_filters::LevelFilter;
+use tracing_subscriber::{
+    layer::SubscriberExt as _, util::SubscriberInitExt as _, EnvFilter,
+};
+
+#[derive(Debug)]
+pub struct TelemetryConfig {
+    syslog_identifier: Option<String>,
+    global_filter: EnvFilter,
+}
+
+impl TelemetryConfig {
+    /// Provides all required arguments for telemetry configuration.
+    /// - `log_identifier` will be used for journald, if appropriate.
+    #[expect(clippy::new_without_default, reason = "may add required args later")]
+    pub fn new() -> Self {
+        Self {
+            syslog_identifier: None,
+            global_filter: EnvFilter::builder()
+                .with_default_directive(LevelFilter::INFO.into())
+                .from_env_lossy(),
+        }
+    }
+
+    /// Enables journald, and uses the provided syslog identifier.
+    ///
+    /// If you run the application in a tty, stderr will be used instead.
+    pub fn with_journald(self, syslog_identifier: &str) -> Self {
+        Self {
+            syslog_identifier: Some(syslog_identifier.to_owned()),
+            ..self
+        }
+    }
+
+    /// Override the global filter to a custom filter.
+    /// Only do this if actually necessary to deviate from the orb's defaults.
+    pub fn with_global_filter(self, filter: EnvFilter) -> Self {
+        Self {
+            global_filter: filter,
+            ..self
+        }
+    }
+
+    /// Initializes the telemetry config. Call this only once, at the beginning of the
+    /// program.
+    ///
+    /// Calling this more than once or when another tracing subscriber is registered
+    /// will cause a panic.
```

Comment:
should we prevent the panic? return an error instead?

Reply:
I think the panic is intended. If we called it more than once we should abort.